core/src/mender-client.c

```diff
@@ -1028,7 +1066,7 @@ mender_filter_provides(mender_artifact_ctx_t *mender_artifact_ctx, mender_key_va
 }
 
 static mender_err_t
-mender_prepare_new_provides(mender_artifact_ctx_t *mender_artifact_ctx, char **new_provides) {
+mender_prepare_new_provides(mender_artifact_ctx_t *mender_artifact_ctx, char **new_provides, char **artifact_name) {
```

Comment:
`const char **artifact_name` to signal that it's not going to contain a newly-allocated string.